src/draw/sw/blend/lv_draw_sw_blend_to_i1.c

```diff
@@ -0,0 +1,1132 @@
+                uint32_t mask_x;
+                for(y = 0; y < h; y++) {
+                    for(mask_x = 0, dest_x = 0, src_x = 0; dest_x < w; mask_x++, dest_x++, src_x++) {
+                        uint8_t src = lv_color16_luminance(src_buf_c16[src_x]);
+                        uint8_t dest = get_bit(dest_buf_u8, dest_x + bit_ofs) * 255;
+                        lv_color_8_8_mix(src, &dest, LV_OPA_MIX2(mask_buf[mask_x], opa));
+                        if(dest > I1_LUM_THRESHOLD) {
+                            set_bit(dest_buf_u8, dest_x + bit_ofs);
+                        }
+                        else {
+                            clear_bit(dest_buf_u8, dest_x + bit_ofs);
+                        }
+                    }
+                    dest_buf_u8 = drawbuf_next_row(dest_buf_u8, dest_stride);
+                    src_buf_c16 = drawbuf_next_row(src_buf_c16, src_stride);
+                    mask_buf += mask_stride;
+                }
+            }
+        }
+    }
+    else {
+        lv_color32_t src_argb;
+        for(y = 0; y < h; y++) {
+            for(src_x = 0, dest_x = 0; src_x < w; src_x++, dest_x++) {
+                src_argb.red = (src_buf_c16[src_x].red * 2106) >> 8;
+                src_argb.green = (src_buf_c16[src_x].green * 1037) >> 8;
+                src_argb.blue = (src_buf_c16[src_x].blue * 2106) >> 8;
+                if(mask_buf == NULL) src_argb.alpha = opa;
+                else src_argb.alpha = LV_OPA_MIX2(mask_buf[src_x], opa);
+
+                blend_non_normal_pixel(dest_buf_u8, dest_x + bit_ofs, src_argb, dsc->blend_mode);
+            }
+            if(mask_buf) mask_buf += mask_stride;
+            dest_buf_u8 = drawbuf_next_row(dest_buf_u8, dest_stride);
+            src_buf_c16 = drawbuf_next_row(src_buf_c16, src_stride);
+        }
+    }
+}
+#endif
+
+static inline void LV_ATTRIBUTE_FAST_MEM blend_non_normal_pixel(uint8_t * dest_buf, int32_t dest_x, lv_color32_t src,
+                                                                lv_blend_mode_t mode)
+{
+    uint8_t res;
+    int32_t src_lumi = lv_color32_luminance(src);
+    uint8_t dest_lumi = get_bit(dest_buf, dest_x) * 255;
+    switch(mode) {
+        case LV_BLEND_MODE_ADDITIVE:
+            res = LV_MIN(dest_lumi + src_lumi, 255);
+            break;
+        case LV_BLEND_MODE_SUBTRACTIVE:
+            res = LV_MAX(dest_lumi - src_lumi, 0);
+            break;
+        case LV_BLEND_MODE_MULTIPLY:
+            res = (dest_lumi * src_lumi) >> 8;
+            break;
+        default:
+            LV_LOG_WARN("Not supported blend mode: %d", mode);
+            return;
+    }
+
+    lv_color_8_8_mix(res, &dest_lumi, src.alpha);
+    if(dest_lumi > I1_LUM_THRESHOLD) {
+        set_bit(dest_buf, dest_x);
+    }
+    else {
+        clear_bit(dest_buf, dest_x);
+    }
+}
+
+static inline void LV_ATTRIBUTE_FAST_MEM lv_color_8_8_mix(const uint8_t src, uint8_t * dest, uint8_t mix)
+{
+
+    if(mix == 0) return;
```

Comment:
TODO: is a branchless version maybe faster?

Reply:
Nope, not really:
![grafik](https://github.com/user-attachments/assets/021e7143-ca48-4e78-a12c-bb436d320bc4)